frontend/src/components/Home.js

```diff
@@ -83,25 +83,36 @@ export default function Home() {
 
   useEffect(() => {
     let url = `${API_HOST}/leads?page=${page}&perpage=${perpage}`;
+    const n_pagesUrl = `${API_HOST}/leads/n_pages`;
     if (search) {
       url += `&search=${search}`;
     }
     const token = localStorage.getItem('partnerFinderToken');
     if (!token) {
       history.push('/login');
     }
+    const headers = {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${token}`,
+    };
     fetch(url, {
-      headers: {
-        'Content-Type': 'application/json',
-        Authorization: `Bearer ${token}`,
-      },
+      headers: headers,
     })
       .then((response) => checkForErrors(response))
       .then((data) => setLeads(data.leads))
       // TODO: create state for error and set state instead of just console.error
       // conditional rendering if there is an error
       .catch((error) => console.error(error.message));
-  }, [page, perpage, search]);
+
+    fetch(n_pagesUrl, {
+      headers: headers,
+    })
+      .then((response) => checkForErrors(response))
+      .then((data) =>
+        setMaxPages(Math.ceil((data.pages * data.query.perpage) / perpage))
```

Comment:
This math is already done on the backend. See my first comment.

```suggestion
        setMaxPages(data.pages)
```